src/main.cpp

```diff
@@ -4230,9 +4231,14 @@ bool static LoadBlockIndexDB()
     pblocktree->ReadFlag("txindex", fTxIndex);
     LogPrintf("%s: transaction index %s\n", __func__, fTxIndex ? "enabled" : "disabled");
 
-    // Check whether we have an address index
-    pblocktree->ReadFlag("addressindex", fAddressIndex);
-    LogPrintf("%s: address index %s\n", __func__, fAddressIndex ? "enabled" : "disabled");
+    // Check whether block explorer features are enabled
+    pblocktree->ReadFlag("insightexplorer", fInsightExplorer);
+    LogPrintf("%s: insight explorer %s\n", __func__, fInsightExplorer ? "enabled" : "disabled");
+    if (fInsightExplorer) {
+        // upstream insight exposes individual index options; we have
+        // chosen to use a single (experimental) flag to control all of them
+        fAddressIndex = true;
```

Comment:
In later PRs, `fSpentIndex` and `fTimestampIndex` will also get set to `true` here.